webapp/src/cloud.ts

```diff
@@ -342,6 +342,23 @@ export function onChangesSynced(changes: Header[]) {
     }
 }
 
+export async function convertCloudToLocal(userId: string) {
+    if (userId) {
+        const localCloudHeaders = workspace.getHeaders(true)
+            .filter(h => h.cloudUserId && h.cloudUserId === userId);
+        const tasks: Promise<void>[] = [];
+        localCloudHeaders.forEach((h) => {
+            // Clear cloud header and re-save the header.
+            delete h.cloudCurrent;
+            delete h.cloudLastSyncTime;
+            delete h.cloudUserId;
+            delete h.cloudVersion;
+            tasks.push(workspace.saveAsync(h));
```

Comment:
saveAsync should be called like: `saveAsync(h, null, true)` in this case. The last parameter, fromCloudSync, indicates this isn't a user change and we shouldn't touch the timestamps on the header.